toqito/state_props/is_abs_ppt.py

```diff
@@ -0,0 +1,128 @@
+"""Check if a quantum state is absolutely PPT."""
+
+import numpy as np
+
+from toqito.matrix_props.is_positive_semidefinite import is_positive_semidefinite
+from toqito.state_opt.symmetric_extension_hierarchy import symmetric_extension_hierarchy
+from toqito.state_props.is_ppt import is_ppt
+from toqito.state_props.is_separable import is_separable
+
+
+def is_abs_ppt(rho: np.ndarray, dim: list[int] = None, max_constraints: int = 2612) -> int:
+    r"""Determine whether a quantum state is absolutely PPT.
+
+    A quantum state :math:`\rho` is absolutely PPT if it remains PPT under any unitary transformation.
+    This function evaluates the condition using:
+
+    - Quick separability checks (to immediately identify separable states)
+    - The PPT property (to test for the PPT condition)
+    - A Gershgorin-type condition (providing a bound based on the eigenvalues)
+    - A symmetric extension hierarchy (for deeper verification when needed)
+
+    .. math::
+        \rho \text{ is absolutely PPT} \iff \forall U,\, (U \rho U^\dagger)^{T_B} \succeq 0.
+
+    Examples
+    ========
+    Example 1: A separable mixed state (trivially PPT)
+
+    >>> import numpy as np
+    >>> from toqito.state_props import is_abs_ppt
+    >>> rho = np.array([[0.5, 0], [0, 0.5]])  # Maximally mixed state
+    >>> is_abs_ppt(rho, [2, 2])
+    1
+
+    Example 2: A Bell state (entangled, not absolutely PPT)
+
+    >>> from toqito.states import bell
+    >>> bell_state = bell(0) @ bell(0).conj().T
+    >>> is_abs_ppt(bell_state, [2, 2])
+    0
+
+    Example 3: A high-dimensional identity matrix (returns -1 for inconclusive cases)
+
+    >>> rho = np.eye(49) / 49
+    >>> is_abs_ppt(rho, [7, 7])
+    -1
+
+    References
+    ==========
+    - MATLAB QETLAB Implementation: Nathaniel Johnston,

```

Comment:
I think we may want to modify the `.bib` file and include a link to this function there. Then, we could just do:

```
    References
    ==========
    .. bibliography::
        :filter: docname in docnames
```